actionpack/lib/action_controller/metal/helpers.rb

```diff
@@ -113,5 +113,10 @@ def all_application_helpers
         all_helpers_from_path(helpers_path)
       end
     end
+
+    # Provides a proxy to access helpers methods from outside the view.
+    def helpers
+      @_helper_proxy ||= view_context
```

Comment:
Because `view_context` is not cached and it working `view_context` is just an implementation detail that I plan to change some point in the future.